org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/codeactions/NoGrammarConstraintsCodeAction.java

```diff
@@ -0,0 +1,127 @@
+package org.eclipse.lemminx.extensions.contentmodel.participants.codeactions;
+
+import java.io.File;
+import java.util.List;
+
+import org.eclipse.lemminx.commons.BadLocationException;
+import org.eclipse.lemminx.commons.CodeActionFactory;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lemminx.dom.DOMElement;
+import org.eclipse.lemminx.extensions.generators.FileContentGeneratorManager;
+import org.eclipse.lemminx.extensions.generators.xml2dtd.DTDGeneratorSettings;
+import org.eclipse.lemminx.extensions.generators.xml2xsd.XMLSchemaGeneratorSettings;
+import org.eclipse.lemminx.services.extensions.ICodeActionParticipant;
+import org.eclipse.lemminx.services.extensions.IComponentProvider;
+import org.eclipse.lemminx.settings.SharedSettings;
+import org.eclipse.lemminx.utils.StringUtils;
+import org.eclipse.lemminx.utils.XMLBuilder;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextDocumentEdit;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+
+public class NoGrammarConstraintsCodeAction implements ICodeActionParticipant {
+
+	@Override
+	public void doCodeAction(Diagnostic diagnostic, Range range, DOMDocument document, List<CodeAction> codeActions,
+			SharedSettings sharedSettings, IComponentProvider componentProvider) {
+		try {
+			DOMElement documentElement = document.getDocumentElement();
+			if (documentElement == null || StringUtils.isEmpty(documentElement.getTagName())) {
+				return;
+			}
+
+			FileContentGeneratorManager generator = componentProvider.getComponent(FileContentGeneratorManager.class);
+			String delimiter = document.lineDelimiter(0);
+			int beforeTagOffset = documentElement.getStartTagOpenOffset();
+			int afterTagOffset = beforeTagOffset + 1 + documentElement.getTagName().length();
+
+			// ---------- XSD
+
+			String schemaURI = getGrammarURI(document.getDocumentURI(), "xsd");
+			String schemaFileName = getFileName(schemaURI);
+			String schemaTemplate = generator.generate(document, sharedSettings, new XMLSchemaGeneratorSettings());
+
+			// xsi:noNamespaceSchemaLocation
+			// Create code action to create the XSD file with the generated XSD content
+			String insertText = " xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\""
+					+ document.getTextDocument().lineDelimiter(0);
+			insertText += " xsi:noNamespaceSchemaLocation=\"" + schemaFileName + "\"";
+			CodeAction noNamespaceSchemaLocationAction = createGrammarFileAndBoundIt(
+					"Bind to a XML Schema with xsi:noNamespaceSchemaLocation", schemaURI, schemaTemplate, insertText,

```

Comment:
Bind to an XML

Reply:
I changed to XSD to be conistent., fixed